runtime/makelib/targets.mk.linux.inc.ftl

```diff
@@ -153,6 +153,16 @@ ifdef USE_PPC_GCC
 endif
 </#if>
 
+<#-- GCC versions greater than 5 default to GNU11 but OpenJ9 uses
+GNU89 inline semantics. -fgnu89-inline option is appended to CFLAGS
+to allow compilation with GCC versions greater than 5 and GNU89 inline
+semantics. Reference - https://gcc.gnu.org/gcc-5/porting_to.html.
+-->
+GCCVERSIONGTEQ5 := $(shell expr `gcc -dumpversion | cut -f1 -d.` \>= 5)
```

Comment:
What happens when gcc 10 ships?  (Not a blocker for merging this now but something to think about while we're looking at this or we'll have to rediscover it later)

Reply:
1. `gcc -dumpversion | cut -f1 -d.` will evaluate to `10`
2. `expr 10 \>= 5` will evaluate to `1`
3. `GCCVERSIONGTEQ5` = `1`
4. `if GCCVERSIONGTEQ5 == 1` then `CFLAGS+=-fgnu89-inline`

Reply:
I misread the `cut` command.

Reply:
We should not be invoking `gcc`, but instead checking if we're using the gcc toolchain and only then invoking `$(CC) -dumpversion`. A system has at most one tool called 'gcc', but the user may have configured their build to use a different version; the test should be based on the selected compiler.

Reply:
I'll put together a PR to address my previous comment.

Reply:
Good catch.  Thanks @keithc-ca

Reply:
Thanks @keithc-ca

Reply:
https://github.com/eclipse/openj9/pull/2205 proposes a fix for this.